---
title: "컴파일러 오류 CS1626"
ms.date: 07/20/2015
ms.prod: .net
ms.technology:
- devlang-csharp
ms.topic: article
f1_keywords:
- CS1626
helpviewer_keywords:
- CS1626
ms.assetid: 3ba03383-eb24-4fd8-bf40-8b0f7d6baf0d
caps.latest.revision: 
author: BillWagner
ms.author: wiwagn
ms.openlocfilehash: 7e991faee8fcd6b8bfbfbbcf77ce243e652a008f
ms.sourcegitcommit: 83dd5ec003e788ccb3e33f3412a7af39ae347646
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 03/15/2018
---
# <a name="compiler-error-cs1626"></a><span data-ttu-id="54fe4-102">컴파일러 오류 CS1626</span><span class="sxs-lookup"><span data-stu-id="54fe4-102">Compiler Error CS1626</span></span>
<span data-ttu-id="54fe4-103">catch 절이 포함된 try 블록의 본문에서는 값을 생성할 수 없습니다.</span><span class="sxs-lookup"><span data-stu-id="54fe4-103">Cannot yield a value in the body of a try block with a catch clause</span></span>  
  
 <span data-ttu-id="54fe4-104">yield 문은 try 블록과 연결된 catch 절이 있는 경우 try 블록에 사용할 수 없습니다.</span><span class="sxs-lookup"><span data-stu-id="54fe4-104">A yield statement is not allowed in a try block if there is a catch clause associated with the try block.</span></span> <span data-ttu-id="54fe4-105">이 오류를 방지하려면 yield 문을 catch 절 밖으로 이동합니다.</span><span class="sxs-lookup"><span data-stu-id="54fe4-105">To avoid this error, move the yield statement out of the catch clause.</span></span>  
  
 <span data-ttu-id="54fe4-106">다음 샘플에서는 CS1626을 생성합니다.</span><span class="sxs-lookup"><span data-stu-id="54fe4-106">The following sample generates CS1626:</span></span>  
  
```csharp  
// CS1626.cs  
using System.Collections;  
  
class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()  
   {  
      try  
      {  
         yield return this;  // CS1626  
      }  
      catch  
      {  
  
      }  
   }  
}  
  
public class CMain  
{  
   public static void Main() { }  
}  
```
